### PR TITLE
Add findOrFail

### PR DESCRIPTION
**Pull request by francoism90**

### CODE REVIEW

File: src/Exceptions/NoPrefixedModelFound.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+use Facade\IgnitionContracts\Solution;
+
+class NoPrefixedModelFound extends Exception implements ProvidesSolution
+{
+    public static function make(string $prefixedId)
+    {
+        return new static("Could not find a prefixed model `{$prefixedId}`");
+    }
+
+    public function getSolution(): Solution
```
Comment (freekmurze) [inline src/Exceptions/NoPrefixedModelFound.php:17]: Nice!

File: src/Models/Concerns/HasPrefixedId.php [hunk 2]
```diff
@@ -32,6 +33,15 @@ public static function findByPrefixedId(string $prefixedId): ?Model
         return static::firstWhere($attributeName, $prefixedId);
     }
 
+    public static function findByPrefixedIdOrFail(string $prefixedId): Model
+    {
+        if (! is_null($model = static::findByPrefixedId($prefixedId))) {
```
Comment (freekmurze) [inline src/Models/Concerns/HasPrefixedId.php:38]: Reverse this condition: throw the exception in the if, return the model at the bottom of the function

**Review by freekmurze — CHANGES_REQUESTED** [review]
Looking good!

File: src/Models/Concerns/HasPrefixedId.php (hunk 2 above)
Comment (francoism90) [reply src/Models/Concerns/HasPrefixedId.php]: @freekmurze I have reversed the condition.

**Comment by riasvdv** [comment]
I'm thinking it might be better to throw Laravel's ModelNotFoundException, as this gets handled into a 404 correctly by default in Laravel. 

```php
throw (new ModelNotFoundException())->setModel(static::class);
```

What do you think @freekmurze?

**Comment by francoism90** [comment]
@riasvdv I was using `ModelNotFoundException` at first, however you cannot provide a proper `getSolution`.

Wouldn't it be possible to extend `NoPrefixedModelFound` with `ModelNotFoundException` instead of `\Exception`?

This would make it possible to listen for both exceptions, right? :)

**Comment by francoism90** [comment]
@freekmurze Thanks for the merge. :)

**Pull request merged**